### PR TITLE
Fix bug where mobile launcher wouldn't find Android devices with hyphens in their product name

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

Fix bug where mobile launcher wouldn't find Android devices with hyphens in their product name #1288 

+ drive-by update to build system asmdef generator

#### Tests

- regex101

#### Documentation

- [x] changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 - Fixed a bug where code generation would happen on every Unity compilation, despite the code generator returning successfully. [#1294](https://github.com/spatialos/gdk-for-unity/pull/1294)
 - Fixed a bug where dotnet output from the code generator would cause exceptions to be thrown. [#1294](https://github.com/spatialos/gdk-for-unity/pull/1294)
+- Fixed a bug where the Mobile Launcher window wouldn't find Android devices that contained hyphens in their name. [#1288](https://github.com/spatialos/gdk-for-unity/issues/1288) [#1296](https://github.com/spatialos/gdk-for-unity/pull/1296)
```
Comment (paulbalaji) [inline CHANGELOG.md]: Or 👇 ?
```suggestion
- Fixed a bug where the Mobile Launcher window wouldn't find Android devices that contained hyphens in their product name. [#1288](https://github.com/spatialos/gdk-for-unity/issues/1288) [#1296](https://github.com/spatialos/gdk-for-unity/pull/1296)
```

Comment (zeroZshadow) [reply CHANGELOG.md]: I'm voting for this one

File: workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/BuildSystemAssemblyGenerator.cs [hunk 2]
```diff
@@ -14,7 +14,12 @@ public static string Generate()
         ""Editor""
     ],
     ""excludePlatforms"": [],
-    ""allowUnsafeCode"": false
+    ""allowUnsafeCode"": false,
```
Comment (zeroZshadow) [inline workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/BuildSystemAssemblyGenerator.cs]: driveby?

Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/BuildSystemAssemblyGenerator.cs]: ye - updated description now
